lib/commands/cleanJobsInSet-3.lua

```diff
@@ -63,7 +65,14 @@ while ((limit <= 0 or deletedCount < limit) and next(jobIds, nil) ~= nil) do
       jobTS = rcall("HGET", jobKey, "timestamp")
       if (not jobTS or jobTS < maxTimestamp) then
         if isList then
-          rcall("LREM", setKey, 0, jobId)
+          if deletionMarker == nil then
```

Comment:
ok, so the risk is to pick a deletion marker that could be a jobId that we are not going to delete, and that is why you pick the first jobId. This is a bit unfortunate, if we could have a short, special value we would 1) skip this "if test" for every item and 2) a potential very large jobId that consumes more time and memory to store and delete. Did you try using as a marker the lua value nil?

Reply:
Alternatively, the value 0 could be used, but then we need to give an error if the user tries to use a custom Id of 0. I am quite sure nobody does this but it should be in place to avoid possible errors.

Reply:
That's a good point about possible large job ids. Since Redis list values are strings, a natural empty value would be the empty string. Luckily, it looks like it's already impossible to add a job with an empty string. Testing with:

```js
queue.add({ some: 'data' }, { jobId: '' })
```

silently ignores the `jobId` and generates a numeric id. This is a consequence of the way the `addJob` Lua script [uses the empty string as a signal that no jobId was passed](https://github.com/OptimalBits/bull/blob/develop/lib/commands/addJob-6.lua#L43-L52).

I'll make that change right away.